src/socket/address.cr

```diff
@@ -50,7 +50,8 @@ class Socket
     @addr6 : LibC::In6Addr?
     @addr4 : LibC::InAddr?
 
-    def initialize(@address : String, @port : Int32)
+    def initialize(address : String, @port : Int32)
+      @address = address
```

Comment:
Why this change?

Ditto a few times below, where inline ivar init is ok.

Reply:
`address` is used in the next line and it should be type `String`. `@address` would be `String?`.

Reply:
Because `@address` is `String?` and the methods/conditions bellow won't work. But `address` is `String` and that works.

Reply:
Oh wow, that's weird to read ^^ but makes sense, ty